### PR TITLE
Shlemen kirill/create tables and views

### PR DESCRIPTION
**Pull request by ShlemenKirill**
Create tables, views, procedures by Admin and Member roles.

### CODE REVIEW

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    UserTask.TaskId AS TaskId,
+    TaskTrack.TaskTrackId AS TaskTrackId,
+    UserProfiles.FirstName AS UserName,
+    --Task.Name AS TaskName,
+    TaskTrack.TrackNote AS TrackNote,
+    TaskTrack.TrackDate AS TrackDate 
+    FROM [UserProfiles] INNER JOIN [UserTask]
+    ON UserProfiles.UserId = UserTask.UserId
+    INNER JOIN [TaskTrack]
+    ON UserTask.UserTaskId = TaskTrack.TaskTrackId
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql:13]: You will still need Task table. Could you leave a comment about it?

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿CREATE VIEW [dbo].[vUserProgress]
+	AS SELECT
+	UserTask.UserId AS UserId,
+    UserTask.TaskId AS TaskId,
+    TaskTrack.TaskTrackId AS TaskTrackId,
+    UserProfiles.FirstName AS UserName,
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql:6]: Here UserName is First and Last names together :)

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi dude.
I left some comment. Fix them and let me know.

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE TABLE [dbo].[TaskState]
+(
+	[StateId] INT NOT NULL PRIMARY KEY,
+	[StateName] NVARCHAR(50) NOT NULL
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql]: I fixed it
```suggestion
[StateName] NVARCHAR(10) NOT NULL
```

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿CREATE TABLE [dbo].[TaskTrack]
+(
+	[TaskTrackId] INT NOT NULL PRIMARY KEY,
+	[UserTaskId] INT NOT NULL,
+	[TrackDate] DATETIME NOT NULL,
+	[TrackNote] INT NOT NULL
+
+	FOREIGN KEY (UserTaskId) REFERENCES TaskState(StateId)
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql]: I fixed it
```suggestion
	FOREIGN KEY (UserTaskId) REFERENCES TaskState(StateId)
	FOREIGN KEY (UserTaskId) REFERENCES UserTask(UserTaskId)
```

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿CREATE TABLE [dbo].[UserTask]
+(
+	[UserTaskId] INT NOT NULL PRIMARY KEY,
+	[TaskId] INT NOT NULL,
+	[UserId] INT NOT NULL,
+	[StateId] INT NOT NULL
+	
+	--FOREIGN KEY (TaskId) REFERENCES Task(TaskID)
+	FOREIGN KEY (UserId) REFERENCES TaskTrack(TaskTrackId)
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql]:
```suggestion
FOREIGN KEY (UserId) REFERENCES UserProfiles(UserId)
```

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql (hunk 1 above)
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql]:
```suggestion
--INNER JOIN [TASK]
--ON UserTask.UserTaskId = Task.TaskId 
```

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql (hunk 2 above)
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql]:
```suggestion
 UserProfiles.FirstName AS UserName,
 (UserProfiles.FirstName + ' ' + UserProfiles.LastName) AS UserName,
```

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿CREATE TABLE [dbo].[TaskTrack]
+(
+	[TaskTrackId] INT NOT NULL PRIMARY KEY,
+	[UserTaskId] INT NOT NULL,
+	[TrackDate] DATETIME NOT NULL,
+	[TrackNote] INT NOT NULL
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql]:
```suggestion
    [TrackNote] NVARCHAR(100) NULL
```

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+	@TaskId int
+AS
+	SELECT UserId FROM UserTask
+	WHERE [UserTask].UserId = @UserId
+	
+	SELECT TaskId FROM UserTask
+	WHERE [UserTask].TaskId = @TaskId
+
+	INSERT INTO TaskState(StateId)
+	VALUES ('Fail')
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql:12]: Maybe this code is right?)
```suggestion
	UPDATE UserTask
	SET StateId = (
	SELECT StateId FROM TaskState
	WHERE [TaskState].StateName = 'Success'
	)
	WHERE [UserTask].TaskId = @TaskId AND [UserTask].UserId = @UserId
```

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql [hunk 8]
```diff
@@ -0,0 +1,10 @@
+﻿
+CREATE PROCEDURE [dbo].[DeleteUser]
+	@userId int	
+AS
+	DELETE FROM UserProfiles 
+	WHERE UserId = @userId
+	
+RETURN 0
```
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql:8]: I need to delete line 8?

[the remark this answers is not in the document]
Comment (PushMeNow) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql:8]: Yeap, you have to. Usually procedure shouldn't have returning value.

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql (hunk 7 above)
[the remark this answers is not in the document]
Comment (PushMeNow) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql:12]: > else insert record with UserId, TaskId and StateId

I don't this code.

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql (hunk 8 above)
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql]: > Yeap, you have to. Usually procedure shouldn't have returning value.
I changed the code

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql (hunk 7 above)
[the remark this answers is not in the document]
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql]:
```suggestion
	CREATE PROCEDURE [dbo].[SetUserTaskAsFail]
	@UserId int,
	@TaskId int
AS
	UPDATE UserTask
	SET StateId = 3
	WHERE [UserTask].TaskId = @TaskId AND [UserTask].UserId = @UserId
```

**Review by Andrey-Gvozdev — APPROVED** [review]

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿CREATE PROCEDURE [dbo].[SetUserTaskAsFail]
+	@UserId int,
+	@TaskId int
+AS
+	IF(EXISTS (SELECT UserId FROM UserTask	WHERE [UserTask].UserId = @UserId) 
+	AND EXISTS (SELECT TaskId FROM UserTask	WHERE [UserTask].TaskId = @TaskId))
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql:6]: You have to join these conditions in one: 
`WHERE UserId = {smth} AND TaskId = {smth}`

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi man!
I did a comment. You have inaccuracy in condition of IF statement.

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql (hunk 9 above)
Comment (ShlemenKirill) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql]: IF EXISTS (SELECT UserId FROM UserTask	WHERE [UserTask].UserId = @UserId AND [UserTask].TaskId = @TaskId)

**Review by PushMeNow — APPROVED** [review]
Approve

**Pull request merged**